### PR TITLE
docs: supplementary health check documents

### PR DESCRIPTION
**Pull request by shuaijinchao**
### What this PR does / why we need it:
Supplementary health check documents.

Fix https://github.com/apache/apisix/issues/3792
Fix https://github.com/apache/apisix/issues/3984

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Comment by shuaijinchao** [comment]
https://github.com/apache/apisix/blob/0ba89d5c0f9dc2863329a8fa6552c380453627cf/apisix/schema_def.lua#L131-L145

**Review by tzssangglass — APPROVED** [review]

**Review by starsz — APPROVED** [review]

File: docs/en/latest/health-check.md [hunk 1]
```diff
@@ -91,12 +91,13 @@ The configures in `checks` are belong to health check, the type of `checks`
...
 * `active`: To enable active health checks, you need to specify the configuration items under `checks.active` in the Upstream object configuration.
-
+  * `active.type`: The type of active health check, supports `http`, `https`, `tcp`, default: `http`.
   * `active.timeout`: Socket timeout for active checks (in seconds), support decimals. For example `1.01` means `1010` milliseconds, `2` means `2000` milliseconds.
-
+  * `active.concurrency`: Number of targets to check concurrently in active health checks, default: `10`.
   * `active.http_path`: The HTTP GET request path used to detect if the upstream is healthy.
   * `active.host`: The HTTP request host used to detect if the upstream is healthy.
   * `active.port`: The customize health check host port (optional), this will override the port in the `upstream` node.
+  * `active.https_verify_certificate`: Whether to check the validity of the SSL certificate of the remote host when performing active health checks using HTTPS. Default: `true`.
```
Comment (spacewander) [inline docs/en/latest/health-check.md:100]: Some fields are still missing, like "http_statuses".

Comment (shuaijinchao) [reply docs/en/latest/health-check.md]: done

**Review by tokers — APPROVED** [review]

File: docs/en/latest/health-check.md [hunk 2]
```diff
@@ -91,21 +91,26 @@ The configures in `checks` are belong to health check, the type of `checks`
...
   * `active.healthy.successes`: The number of success times to determine the target is healthy, the minimum is 1.
 
   The threshold fields of  `unhealthy` are:
   * `active.unhealthy.interval`: Interval between health checks for unhealthy targets (in seconds), the minimum is 1.
-  * `active.unhealthy.http_failures`: The number of http failures times to determine the target is unhealthy, the minimum is 1.
-  * `active.req_headers`: Additional request headers. Array format, so you can fill in multiple headers.
+  * `active.unhealthy.http_statuses`: When using an HTTP request to check the health status of a node, if the response status code matches the set status code, the node is set to the `unhealthy` status, array format, default: `[429, 404, 500, 501, 502, 503, 504, 505]`。
+  * `active.unhealthy.http_failures`: Determine the number of unhealthy http request failures on the target node, default: `5`.
+  * `active.unhealthy.tcp_failures`: Determine the number of unhealthy tcp request failures on the target node, default: `5`.
```
Comment (spacewander) [inline docs/en/latest/health-check.md:112]:
```suggestion
  * `active.unhealthy.tcp_failures`: Determine the number of unhealthy tcp request failures on the target node, default: `2`.
```

**Review by spacewander — COMMENTED** [review]
Could you add the default value for the other fields, and add 'optional' for fields which are optional? Better to update the passive part of this doc.

File: docs/en/latest/health-check.md (hunk 2 above)
Comment (shuaijinchao) [reply docs/en/latest/health-check.md]: done

File: docs/en/latest/health-check.md [hunk 3]
```diff
@@ -35,7 +34,34 @@ There won't be any health check if an upstream is configured but isn't in used.
...
+| upstream.checks.active.unhealthy.tcp_failures   | Active check (unhealthy node)  | integer    | `1` to `254`         | 2                                                                                             | Active check (unhealthy node) TCP type check, determine the number of times that the node is not healthy.            |
+| upstream.checks.active.unhealthy.timeouts       | Active check (unhealthy node)  | integer    | `1` to `254`         | 3                                                                                             | Active check (unhealthy node) to determine the number of timeouts for unhealthy nodes.                              |
+| upstream.checks.passive.healthy.http_statuses   | Passive check (healthy node)   | array      | `200` to `599`       | [200, 201, 202, 203, 204, 205, 206, 207,208, 226, 300, 301, 302, 303, 304, 305,306, 307, 308] | Passive check (healthy node) HTTP or HTTPS type check, the HTTP status code of the healthy node.                     |
```
Comment (spacewander) [inline docs/en/latest/health-check.md:57]:
```suggestion
| upstream.checks.passive.healthy.http_statuses   | Passive check (healthy node)   | array      | `200` to `599`       | [200, 201, 202, 203, 204, 205, 206, 207, 208, 226, 300, 301, 302, 303, 304, 305, 306, 307, 308] | Passive check (healthy node) HTTP or HTTPS type check, the HTTP status code of the healthy node.                     |
```

**Review by spacewander — APPROVED** [review]

File: docs/en/latest/health-check.md (hunk 3 above)
Comment (shuaijinchao) [reply docs/en/latest/health-check.md]: done

**Review by spacewander — APPROVED** [review]

**Pull request merged**